Building on latest msys2 chain on windows

@cvvergara I was writing up instructions on how to build on windows with msys2 and mingw64 and ran into a stumbling block

Here are the [instructions](https://github.com/pgRouting/pgrouting/wiki/Building-on-windows-with-msys2-and-mingw64-(WIP))

all goes smoothly until the build step, which errors in 


> cd build
> cmake -G"Unix Makefiles" ..
> cmake --build .
> -- Setting build type to 'Release' as none was specified.
> -- CMAKE_BUILD_TYPE Release
> -- The C compiler identification is GNU 12.2.0
> -- The CXX compiler identification is GNU 12.2.0
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Check for working C compiler: C:/Msys2pgRouting/msys2/mingw64/bin/cc.exe - skipped
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Check for working CXX compiler: C:/Msys2pgRouting/msys2/mingw64/bin/c++.exe - skipped
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- Found Git: C:/Msys2pgRouting/msys2/usr/bin/git.exe (found version "2.39.1")
> -- DOXYGEN_MINIMUM_VERSION=1.7
> -- SPHINX_MINIMUM_VERSION=4.0
> -- POSTGRESQL_MINIMUM_VERSION=9.2.0
> -- BOOST_MINIMUM_VERSION=1.56.0
> -- POSTGIS_MINIMUM_VERSION=2.0.0
> -- Found Boost: C:/Msys2pgRouting/msys2/mingw64/lib/cmake/Boost-1.81.0/BoostConfig.cmake (found suitable version "1.81.0", minimum required is "1.56.0")
> -- Found Perl: C:/Msys2pgRouting/msys2/usr/bin/perl.exe (found version "5.36.0")
> -- POSTGRESQL_PG_CONFIG is C:/Msys2pgRouting/msys2/mingw64/bin/pg_config.exe
> -- POSTGRESQL_EXECUTABLE is C:/Msys2pgRouting/msys2/mingw64/bin/postgres.exe
> -- POSTGRESQL_VERSION_STRING in FindPostgreSQL.cmake is PostgreSQL 15.1
> -- POSTGRESQL_INCLUDE_DIR: C:/Msys2pgRouting/msys2/mingw64/include/postgresql/server
> -- POSTGRESQL_LIBRARIES: C:/MSYS2P~1/msys2/mingw64/lib
> -- POSTGRESQL_VERSION_STRING=PostgreSQL 15.1
> -- POSTGRESQL_VERSION=15.1
> -- PGSQL_VERSION=151
> -- LIBRARY_INSTALL_PATH C:/MSYS2P~1/msys2/mingw64/lib/POSTGR~1
> -- Performing Test C_COMPILER_SUPPORTS_ROUNDING_MATH
> -- Performing Test C_COMPILER_SUPPORTS_ROUNDING_MATH - Success
> -- Performing Test CXX_COMPILER_SUPPORTS_ROUNDING_MATH
> -- Performing Test CXX_COMPILER_SUPPORTS_ROUNDING_MATH - Success
> -- Configuring done
> -- Generating done
> -- Build files have been written to: C:/projects/pgRouting/build
> [  1%] Building C object src/ordering/CMakeFiles/ordering.dir/cuthillMckeeOrdering.c.obj
> [  1%] Building CXX object src/ordering/CMakeFiles/ordering.dir/cuthillMckeeOrdering_driver.cpp.obj
> In file included from C:/projects/pgRouting/include/ordering/cuthillMckeeOrdering.hpp:46,
>                  from C:/projects/pgRouting/src/ordering/cuthillMckeeOrdering_driver.cpp:38:
> C:/projects/pgRouting/include/ordering/cuthillMckeeOrdering.hpp: In member function 'std::vector<II_t_rt> pgrouting::functions::CuthillMckeeOrdering<G>::c
> uthillMckeeOrdering(G&)':
> C:/projects/pgRouting/include/cpp_common/interruption.h:75:15: error: there are no arguments to 'UNBLOCKED_SIGNAL_QUEUE' that depend on a template paramet
> er, so a declaration of 'UNBLOCKED_SIGNAL_QUEUE' must be available [-fpermissive]
>    75 |     (unlikely(UNBLOCKED_SIGNAL_QUEUE()) ? pgwin32_dispatch_queued_signals() : 0, \
>       |               ^~~~~~~~~~~~~~~~~~~~~~
> C:/projects/pgRouting/include/cpp_common/interruption.h:63:39: note: in definition of macro 'unlikely'
>    63 | #define unlikely(x) __builtin_expect((x) != 0, 0)
>       |                                       ^
> C:/projects/pgRouting/include/cpp_common/interruption.h:82:13: note: in expansion of macro 'INTERRUPTS_PENDING_CONDITION'
>    82 |         if (INTERRUPTS_PENDING_CONDITION()) \
>       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
> C:/projects/pgRouting/include/ordering/cuthillMckeeOrdering.hpp:108:10: note: in expansion of macro 'CHECK_FOR_INTERRUPTS'
>   108 |          CHECK_FOR_INTERRUPTS();
>       |          ^~~~~~~~~~~~~~~~~~~~
> C:/projects/pgRouting/include/cpp_common/interruption.h:75:15: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an und
> eclared name is deprecated)
>    75 |     (unlikely(UNBLOCKED_SIGNAL_QUEUE()) ? pgwin32_dispatch_queued_signals() : 0, \
>       |               ^~~~~~~~~~~~~~~~~~~~~~
> C:/projects/pgRouting/include/cpp_common/interruption.h:63:39: note: in definition of macro 'unlikely'
>    63 | #define unlikely(x) __builtin_expect((x) != 0, 0)
>       |                                       ^
> C:/projects/pgRouting/include/cpp_common/interruption.h:82:13: note: in expansion of macro 'INTERRUPTS_PENDING_CONDITION'
>    82 |         if (INTERRUPTS_PENDING_CONDITION()) \
>       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
> C:/projects/pgRouting/include/ordering/cuthillMckeeOrdering.hpp:108:10: note: in expansion of macro 'CHECK_FOR_INTERRUPTS'
>   108 |          CHECK_FOR_INTERRUPTS();
>       |          ^~~~~~~~~~~~~~~~~~~~
> C:/projects/pgRouting/include/ordering/cuthillMckeeOrdering.hpp: In instantiation of 'std::vector<II_t_rt> pgrouting::functions::CuthillMckeeOrdering<G>::
> cuthillMckeeOrdering(G&) [with G = pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, pgrouting::Basic_v
> ertex, pgrouting::Basic_edge>, pgrouting::Basic_vertex, pgrouting::Basic_edge>]':
> C:/projects/pgRouting/src/ordering/cuthillMckeeOrdering_driver.cpp:63:64:   required from 'std::vector<II_t_rt> cuthillMckeeOrdering(G&) [with G = pgrouti
> ng::graph::Pgr_base_graph<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, pgrouting::Basic_vertex, pgrouting::Basic_edge>, pgrouting::
> Basic_vertex, pgrouting::Basic_edge>]'
> C:/projects/pgRouting/src/ordering/cuthillMckeeOrdering_driver.cpp:117:39:   required from here
> C:/projects/pgRouting/include/cpp_common/interruption.h:75:37: error: 'UNBLOCKED_SIGNAL_QUEUE' was not declared in this scope
>    75 |     (unlikely(UNBLOCKED_SIGNAL_QUEUE()) ? pgwin32_dispatch_queued_signals() : 0, \
>       |                                     ^
> C:/projects/pgRouting/include/cpp_common/interruption.h:63:39: note: in definition of macro 'unlikely'
>    63 | #define unlikely(x) __builtin_expect((x) != 0, 0)
>       |                                       ^
> C:/projects/pgRouting/include/cpp_common/interruption.h:82:13: note: in expansion of macro 'INTERRUPTS_PENDING_CONDITION'
>    82 |         if (INTERRUPTS_PENDING_CONDITION()) \
>       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
> C:/projects/pgRouting/include/ordering/cuthillMckeeOrdering.hpp:108:10: note: in expansion of macro 'CHECK_FOR_INTERRUPTS'
>   108 |          CHECK_FOR_INTERRUPTS();
>       |          ^~~~~~~~~~~~~~~~~~~~
> make[2]: *** [src/ordering/CMakeFiles/ordering.dir/build.make:92: src/ordering/CMakeFiles/ordering.dir/cuthillMckeeOrdering_driver.cpp.obj] Error 1
> make[1]: *** [CMakeFiles/Makefile2:2279: src/ordering/CMakeFiles/ordering.dir/all] Error 2
> make: *** [Makefile:136: all] Error 2

--- a/include/cpp_common/interruption.h
+++ b/include/cpp_common/interruption.h
@@ -35,54 +35,8 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
 extern "C" {
 
-#ifdef _MSC_VER
-
 #include <postgres.h>
 #include <miscadmin.h>
-
-#else
-    /*
-     * Instead of including all the files
-     * copy/paste what is used
-     * IMPORTANT: this copy/paste might depend on the postgreSQL version
-     */
-#include <signal.h>
-
-    /*
-     * https://doxygen.postgresql.org/c_8h.html#a166c1d950e659804f0e3247aad99a81f
-     */
-#define PGDLLIMPORT
-
-    /*
-     * https://doxygen.postgresql.org/miscadmin_8h_source.html
-     */
-    extern PGDLLIMPORT volatile sig_atomic_t InterruptPending;
-    extern void ProcessInterrupts(void);
-
-#if __GNUC__ >= 3
-#define unlikely(x) __builtin_expect((x) != 0, 0)
-#else
-#define unlikely(x) ((x) != 0)
-#endif
-
-    /* Test whether an interrupt is pending */
-#ifndef WIN32
-#define INTERRUPTS_PENDING_CONDITION() \
-    (unlikely(InterruptPending))
-#else
-    extern void pgwin32_dispatch_queued_signals(void);
-#define INTERRUPTS_PENDING_CONDITION() \
-    (unlikely(UNBLOCKED_SIGNAL_QUEUE()) ? pgwin32_dispatch_queued_signals() : 0, \
-     unlikely(InterruptPending))
-#endif
-
-    /* Service interrupt, if one is pending and it's safe to service it now */
-#define CHECK_FOR_INTERRUPTS() \
-    do { \
-        if (INTERRUPTS_PENDING_CONDITION()) \
-        ProcessInterrupts(); \
-    } while (0)
-#endif
 
 }
 

